core/dbt/compilation.py

```diff
@@ -465,10 +470,16 @@ def add_test_edges(self, linker: Linker, manifest: Manifest) -> None:
                     # node(s) it depends on.
                     test_depends_on = set(manifest.nodes[upstream_test].depends_on_nodes)
 
+                    all_depends_on_node = set()
+                    for depends_on_node in test_depends_on:
```

Comment:
This might be expensive to run

Reply:
Good callout. This is worth testing with a very big project (e.g. [this one](https://github.com/dbt-labs/dbt-core/tree/main/performance/projects/01_2000_simple_models)). `dbt build` is already much slower during graph compilation than other tasks, because it alone adds extra test edges (this method).

Reply:
I just tested locally using that perf project, and found that this method takes:
- 3 seconds to run on current `main`
- multiple minutes with the change on this branch

Reply:
When we had a similar performance issue earlier (also looping over everything for every node) we switched to using the pre-built dag (parent_map, child_map). Maybe we could build a specialized dag including tests?

Reply:
Ha mine doesn't even finish on it. And switching to using the (parent_map, child_map) is still slow. I tried out limiting the bfs search depth to be 1 and found that speeds things up a lot on big projects and still is correct.
Though in the original issue there is still a situation I think makes leaving current behavior desirable.